lib/node_modules/@stdlib/utils/async/if-else/docs/types/index.d.ts

```diff
@@ -61,6 +61,7 @@ type PredicateCallback =  PredicateNullary | PredicateUnary | PredicateBinary;
 */
 type Predicate = ( clbk: PredicateCallback ) => void;
 
+type ResultFunction<T,U> = T extends U ? T : U;
```

Comment:
Notice that this would incorrectly infer one of the types as the `return` value type in case we have `x` and `y` values of different types. I propose using a union type in this situation:

```suggestion
/**
 * Determines the relationship between two types `T` and `U`, and returns:
 * - `T` if `T` is assignable to `U`,
 * - `U` if `U` is assignable to `T`,
 * - A union of `U | T` otherwise.
 */
type ResultFunction<T,U> = T extends U ? T : U extends T ? U : U | T;

```

Reply:
Ah right, that makes sense. Thank you for bringing it to the light :)